src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1376,15 +1376,24 @@ private bool IsQueryMatch(string query, string symbolName)
 
         #region Event Handlers
 
-        private FoldingRange[] Fold(
-            string documentUri)
+        private FoldingRange[] Fold(string documentUri)
         {
             // TODO Should be using dynamic registrations
             if (!this.currentSettings.CodeFolding.Enable) { return null; }
+
+            // Avoid crash when using untitled: scheme or any other scheme where the document doesn't
+            // have a backing file.  https://github.com/PowerShell/vscode-powershell/issues/1676
+            // Perhaps a better option would be to parse the contents of the document as a string
+            // as opposed to reading a file but the senario of "no backing file" probably doesn't
+            // warrant the extra effort.
+            ScriptFile scriptFile = editorSession.Workspace.GetFile(documentUri);
```

Comment:
I recently added a TryGetFile function that would do the soft check:

https://github.com/PowerShell/PowerShellEditorServices/blob/99f01efedf1f4097b123365d49040ebb61b3a92c/src/PowerShellEditorServices/Workspace/Workspace.cs#L142

maybe that's better to use here?

Reply:
We could (perhaps should) change the call site but there are 23 references to `GetFile()` so maybe we should still try to make `GetFile()` a little more robust.  Either that OR we nuke `GetFile()` and replace all 23 references with calls to `TryGetFile()`.  Longer term, that may be the better strategy because it makes it clearer you may not get a file back.  `GetFile()` behaves similarly except that it returns a `null` when there is no file to return - which is way less obvious and easier to ignore the `null` case - kind of like ignoring error return codes in C code.  :-)

Reply:
Actually I don't think `GetFile()` handle's File exceptions:

https://github.com/PowerShell/PowerShellEditorServices/blob/99f01efedf1f4097b123365d49040ebb61b3a92c/src/PowerShellEditorServices/Workspace/Workspace.cs#L118-L119

If `FileStream` or `StreamReader` throws an exception, that will bubble up I'm pretty sure, no?

Where as `TryGetFile` actually does catch them.

I think we should _eventually_ move to using `TryGetFile` everywhere it makes sense... but we can call that out of scope for this PR if you'd like.

Reply:
I'll make the change to use `TryGetFile()` later tonight.  Getting ready to head out to see Aquaman right now.  :-)

Reply:
Have a great time! I want to hear the full @rkeithhill review 😄

Reply:
OK, updated to use `TryGetFile()` instead - reverted change to `GetFile()`.  Although I did an "important" note on `GetFile()` that shows up in Intellisense that recommends `TryGetFile()` instead.  Also, fix an incorrect (copy/paste) log message.

P.S. Aquaman was pretty good. Not as good as Wonder Woman IMO but way better than Dawn of Pustice.  Then again, I've been a Jason Momoa fan since Stargate Atlantis.  :-)